contributing: update minimum supported Rust policy (#1431)
PyO3/pyo3: Contributing.md

Tests run with all supported Python versions with the latest stable Rust compiler, as well as for Python 3.9 with the minimum supported Rust version.

## Python and Rust version support policy

PyO3 aims to keep sufficient compatibility to make packaging Python extensions built with PyO3 feasible on most common package managers.

> To keep package maintainer's lives simpler, PyO3 will commit, wherever possible, to only adjust minimum supported Rust and Python versions at the same time. This release will come roughly once per year, after the oldest supported Python version reaches its end-of-life. (Check https://endoflife.date/python for a clear timetable on these.)

Reviewer: ```suggestion
To keep package maintainers' lives simpler, PyO3 will commit, wherever possible, to only adjust minimum supported Rust and Python versions at the same time. This bump will come in a new `0.x` release, roughly once per year, after the oldest supported Python version reaches its end-of-life. (Check https://endoflife.date/python for a clear timetable on these.)
```

Author: Hmm do you think `in a new 0.x release` implies we won't make 0.x releases at other times in the year?

Reviewer: Right, it might be read that way.  maybe `will only come in a 0.x release`?

Author: I like that, let's go for that.

Below are guidelines on what compatibility all PRs are expected to deliver for each language.

> ## Python

Reviewer 2: You meant:
```suggestion
### Python
```
?

Author: 👍

PyO3 supports all oficially supported Python versions, as well as the latest PyPy3 release. All of these versions are tested in CI.

> ## Rust

Reviewer 2: Same as above,
```suggestion
### Rust
```

Author: Thanks!
